lib/charms/mongodb/v1/shards_interface.py

```diff
@@ -145,6 +145,13 @@ def _on_relation_joined(self, event):
 
         self.database_provides.update_relation_data(event.relation.id, relation_data)
 
+    def __handle_relation_not_feasible(self, event: EventBase):
+        if self.charm.status.is_status_related_to_mismatched_revision(self.charm.unit.status.name):
```

Comment:
Given this method only performs a check/deferral - would it make sense to directly put the logic in `pass_sanity_hook_checks`? Or are you expecting this method to perform additional logic in the future?

Reply:
I'm not expecting it to perform additional logic, I moved it from `pass_sanity_hook_checks` mainly to remove some nesting in the main function after [this](https://github.com/canonical/mongodb-operator/pull/435#discussion_r1687577879) comment from Mia.
I'm fine with both, I just find this code easier to read.

In the future, we should improve this `pass_sanity_hook_checks` and the way the `is_relation_feasible` works because `is_relation_feasible` (which looks like a `GET` function) does sets the status and it would be better that this function either returns nothing (AKA all is good) or a status that we can then handle and set. This helps decorrelating the functionality and makes a code easier to read and understand.